### PR TITLE
support FiT training/inference

### PR DESCRIPTION
**Pull request by zhtmike**
This PR includes:
- Add FiT model (training pipeline / inference pipeline / SwiGLU / 2D RoPE / VisionNTK). 
- The model is under training and the result seems promising.
- Move ImageNet Dataset in DiT to separate file for better look.
- Add `--imagegrid` support in DiT to show images in grid format.
- Align the CenterCrop in DiT with the official repo.
- Move some universal functions in `models/dit.py` to `utils.py` for easier reuse for other models.
- Refactor and add positional embedding (2D sinusoidal embedding and 2D Rotatery Embedding) in `models/modules`, for easier reuse for others models.
- Minor formating in DiT folder

TODO:
- Update README and benchmark once training is finished.

# What does this PR do?

Fixes # (issue)

Adds # (feature)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [x] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [x] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

File: examples/fit/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+python sample.py --imagegrid True
+
+```
+You can also adjust the image size by add the flag `--image_height` and `--image_width`. For example, you can run
+```bash
+python sample.py --imagegrid True --image_height 320 --image_width 160
+```
+to generate image with 160x320 size.
+
+## Training with ImageNet format
```
Comment (wtomin) [inline examples/fit/README.md:26]: Before introducing the training scripts for ImageNet dataset, please also introduce where to download ImageNet dataset, and the folder structure of the downloaded data.

File: examples/fit/pipelines/infer_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+from mindspore import Tensor, ops
+
+from mindone.models.modules import get_2d_sincos_pos_embed, precompute_freqs_cis_2d
+
+
+class FiTInferPipeline:
+    """
+
+    Args:
+        dit (nn.Cell): A `FiT` to denoise the encoded image latents.
```
Comment (wtomin) [inline examples/fit/pipelines/infer_pipeline.py:16]: correct `dit` to `fit`, or more general name `model`

File: mindone/models/fit.py [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+    """
+    A FiT block with adaptive layer norm zero (adaLN-Zero) conditioning.
+    """
+
+    def __init__(
+        self,
+        hidden_size: int,
+        num_heads: int,
+        mlp_ratio: float = 4.0,
+        ffn: str = "swiglu",
```
Comment (hadipash) [inline mindone/models/fit.py:184]: It will be good to list all possible values:
`ffn: Literal["swiglu", "mlp"] = "swiglu" `

**Review by hadipash — APPROVED** [review]

File: examples/fit/README.md (hunk 1 above)
Comment (zhtmike) [reply examples/fit/README.md:33]: Done.

File: examples/fit/pipelines/infer_pipeline.py (hunk 2 above)
Comment (zhtmike) [reply examples/fit/pipelines/infer_pipeline.py]: Fixed. Thx

File: mindone/models/fit.py (hunk 3 above)
Comment (zhtmike) [reply mindone/models/fit.py:184]: python 3.7 does not support literal, but so far mindspore still support 3.7 version..

Comment (hadipash) [reply mindone/models/fit.py:184]: Can do this way:
https://github.com/mindspore-lab/mindone/blob/ead5e008bac80fa5faa9e798ba03d5d5aab268a7/mindone/env/init.py#L5-L8
Sooner or later support for 3.7 will be dropped.

Comment (zhtmike) [reply mindone/models/fit.py]: great! fixed.

**Review by wtomin — APPROVED** [review]

File: examples/dit/configs/training/class_cond_train.yaml [hunk 4]
```diff
@@ -11,13 +11,13 @@ model_name: "DiT-XL/2"
...
 sd_scale_factor: 0.18215
-enable_flash_attention: True
+enable_flash_attention: False
 use_fp16: True
 
 # data setting
 imagenet_format: True
 data_path: imagenet_samples/images/
-dataset_sink_mode: True
+dataset_sink_mode: False
```
Comment (SamitHuang) [inline examples/dit/configs/training/class_cond_train.yaml:20]: Is it runnable in data sink mode?

**Review by SamitHuang — APPROVED** [review]
lgtm

File: examples/dit/configs/training/class_cond_train.yaml (hunk 4 above)
Comment (zhtmike) [reply examples/dit/configs/training/class_cond_train.yaml:20]: yes. But so far the benchmark and training is based on non-data sink model. Can be turn on together with flash attention later

Comment (SamitHuang) [reply examples/dit/configs/training/class_cond_train.yaml:20]: great. let's update the training throughput later with FA and DS

**Review by CaitinZhao — APPROVED** [review]

**Pull request merged**